modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/FSMCallerImpl.java

```diff
@@ -665,6 +676,27 @@ private void doSnapshotLoad(final LoadSnapshotClosure done) {
             setError(e);
             return;
         }
+
+        // JRaft tests use such strange metas, so we have to protect... in production, these are never null.
```

Comment:
I didn't get this comment

Reply:
Take a look at `FSMCallerTest:166` (`testOnSnapshotLoad()`). The test constructs a snapshot meta without peers/learners info (so `peers()` and `learners()` return `null`). Here, the `if` protects from such strange metas.

In production code, `peers()` and `learners()` on `SnapshotMeta` never return `null`.

Reply:
I clarified the comment a bit.